zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -477,25 +591,67 @@ benchmarks! {
     }: redeem_shares(RawOrigin::Signed(caller), market_id)
 
     reject_market {
+        let c in 0..63;
+        let o in 0..63;
+
         let (_, market_id) = create_market_common::<T>(
             MarketCreation::Advised,
             MarketType::Categorical(T::MaxCategories::get()),
             ScoringRule::CPMM
         )?;
+
+        let range_start = T::MinSubsidyPeriod::get();
+        let range_end = T::MaxSubsidyPeriod::get();
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            market.period = MarketPeriod::Timestamp(range_start..range_end);
+            Ok(())
+        })?;
+
+        for i in 0..o {
+            MarketIdsPerOpenTimeFrame::<T>::try_mutate(
+                Pallet::<T>::calculate_time_frame_of_moment(range_start),
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
+        }
+
+        for i in 0..c {
+            MarketIdsPerCloseTimeFrame::<T>::try_mutate(
+                Pallet::<T>::calculate_time_frame_of_moment(range_end),
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
+        }
+
         let reject_origin = T::RejectOrigin::successful_origin();
         let call = Call::<T>::reject_market { market_id };
     }: { call.dispatch_bypass_filter(reject_origin)? }
 
     report {
+        let m in 0..63;
+
         let (caller, market_id) = create_market_common::<T>(
             MarketCreation::Permissionless,
             MarketType::Categorical(T::MaxCategories::get()),
             ScoringRule::CPMM
         )?;
+
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            // ensure to check the origin
```

Comment:
I don't get this comment. What do you mean by this?

Reply:
I meant, that the `range.start` should be now, to ensure the heaviest path gets executed. Good point, comment edited.

Reply:
https://github.com/zeitgeistpm/zeitgeist/blob/17bdbafbd537de84b03a52afb011838f139e18c3/zrml/prediction-markets/src/benchmarks.rs#L632-L644